rosbag2_test_common/include/rosbag2_test_common/temporary_directory_fixture.hpp

```diff
@@ -17,26 +17,61 @@
 
 #include <gmock/gmock.h>
 
+#include <filesystem>
 #include <string>
 
-#include "rcpputils/filesystem_helper.hpp"
+#ifdef _WIN32
+#  include <io.h>
+#endif
 
 using namespace ::testing;  // NOLINT
 
 namespace rosbag2_test_common
 {
 
+/**
+ * @brief std::filesystem version of create_temp_directory from rcpputils/src/filesystem_helper.cpp
+ */
+std::filesystem::path create_temp_directory(
+  const std::string & base_name,
+  const std::filesystem::path & parent_path = std::filesystem::temp_directory_path())
+{
+  const auto template_path = base_name + "XXXXXX";
+  std::string full_template_str = (parent_path / template_path).generic_string();
+  std::filesystem::create_directories(parent_path);
```

Comment:
std::filesystem::create_directories could fail and return false.

```suggestion
  if(!std::filesystem::create_directories(parent_path)) {
    std::error_code ec{errno, std::system_category()};
    errno = 0;
    throw std::system_error(ec, "could not create the parent directory");
  }
```

Reply:
It works same as before no need to handle `false` return code.

`std::filesystem::create_directories(parent_path)` will:
1. return true if has actually created directories
2. return false if directory already exists
3. throw exception in other cases (e.g. permisssion denied or not a directory)

This behaviour aligned with the version from rcpputils. Except may be that std::filesystem_error and not std::system_error is thrown. This function only used in tests, so I think we can skip adding tests.

Reply:
i see! thanks for the info!